indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/autoscaler/CostBasedAutoScalerConfig.java

```diff
@@ -0,0 +1,386 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.seekablestream.supervisor.autoscaler;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import org.apache.druid.indexing.overlord.supervisor.Supervisor;
+import org.apache.druid.indexing.overlord.supervisor.SupervisorSpec;
+import org.apache.druid.indexing.overlord.supervisor.autoscaler.SupervisorTaskAutoScaler;
+import org.apache.druid.indexing.seekablestream.supervisor.SeekableStreamSupervisor;
+import org.apache.druid.java.util.emitter.service.ServiceEmitter;
+
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+/**
+ * Configuration for cost-based auto-scaling of seekable stream supervisor tasks.
+ * Uses a cost function combining lag and idle time metrics to determine optimal task counts.
+ * Task counts are constrained to be factors/divisors of the partition count.
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CostBasedAutoScalerConfig implements AutoScalerConfig
+{
+  private static final long DEFAULT_METRICS_COLLECTION_INTERVAL_MILLIS = 3 * 60 * 1000; // 3 minutes
+  private static final long DEFAULT_SCALE_ACTION_START_DELAY_MILLIS = 10 * 60 * 1000; // 10 minutes
+  private static final long DEFAULT_SCALE_ACTION_PERIOD_MILLIS = 15 * 60 * 1000; // 15 minutes
+  private static final long DEFAULT_MIN_TRIGGER_SCALE_ACTION_FREQUENCY_MILLIS = 1200000; // 20 minutes
+  private static final double DEFAULT_LAG_WEIGHT = 0.25;
+  private static final double DEFAULT_IDLE_WEIGHT = 0.75;
+
+  private final boolean enableTaskAutoScaler;
+  private final int taskCountMax;
+  private final int taskCountMin;
+  private Integer taskCountStart;
+  private final long minTriggerScaleActionFrequencyMillis;
+  private final Double stopTaskCountRatio;
+  private final long metricsCollectionIntervalMillis;
+  private final long scaleActionStartDelayMillis;
+  private final long scaleActionPeriodMillis;
+
+  private final double lagWeight;
+  private final double idleWeight;
+
+  @JsonCreator
+  public CostBasedAutoScalerConfig(
+      @JsonProperty("taskCountMax") Integer taskCountMax,
+      @JsonProperty("taskCountMin") Integer taskCountMin,
+      @Nullable @JsonProperty("enableTaskAutoScaler") Boolean enableTaskAutoScaler,
+      @Nullable @JsonProperty("taskCountStart") Integer taskCountStart,
```

Comment:
Based on the discussions on the other PR #18745 , I wonder if we should just skip adding a `taskCountStart` in this auto-scaler config.